src/trunnel/sendme.trunnel

```diff
@@ -1,18 +1,19 @@
 /* This file contains the SENDME cell definition. */
 
+/* v1 digest length in bytes. */
+const TRUNNEL_SENDME_V1_DIGEST_LEN = 20;
+
+/* SENDME cell declaration. */
 struct sendme_cell {
   /* Version field. */
   u8 version IN [0x00, 0x01];
 
-  /* The data content depends on the version. */
+  /* Length of data contained in this cell. */
   u16 data_len;
-  u8 data[data_len];
-}
 
-/* SENDME version 0. No data. */
-
-/* SENDME version 1. Authenticated with digest. */
-struct sendme_data_v1 {
-  /* A 4 bytes digest. */
-  u8 digest[4];
+  /* The data content depends on the version. */
+  union data[version] with length data_len {
+    0x00: ignore;
+    0x01: u8 v1_digest[TRUNNEL_SENDME_V1_DIGEST_LEN];
+  };
```

Comment:
Do we want parsing to fail if the version is 2 or more?  If not we should add a default: case to the union.

Reply:
I believe we do want to fail. A SENDME you can't parse (most likely unknown version) means you can't continue and thus the circuit is closed. This is the sendme cell validation code:

```
    if (sendme_cell_parse(&cell, cell_payload, cell_payload_len) < 0) {
      log_fn(LOG_PROTOCOL_WARN, LD_PROTOCOL,
             "Unparseable SENDME cell received. Closing circuit.");
      goto invalid;
    }
```